core/cont/src/TClonesArray.cxx

```diff
@@ -444,10 +446,23 @@ void TClonesArray::Clear(Option_t *option)
    TObjArray::Clear();
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Clear a slot in the array with option opt, later nullify it and call Compress
+/// \param index the index of the array to clear, it must be in range of length of fCont, range-check is not performed
+/// \param opt the option passed to the Clear function
+void TClonesArray::ClearSlot(Int_t index, Option_t *opt)
+{
+   if (fCont[index]) {
+      fCont[index]->Clear(opt);
+      fCont[index] = nullptr;
+      Compress();
```

Comment:
Apriori we want the expansive part of the operation (`Compress`) to be left to the user's discretion.  In addition we need to record that the content was updated.
```suggestion
      Changed();
```